src/app/product.service.ts

```diff
@@ -10,4 +10,12 @@ export class ProductService {
   create(product) {
     return this.database.list('/products').push(product);
   }
+
+  getAll() {
+    return this.database.list('/products').snapshotChanges();
```

Comment:
do you need snapshot change here? it making for some heavy lifting in your html product.payload.val().price

Reply:
If I take it out and try to access my product fields like below, I don't render any data:

`<tr *ngFor="let product of productsObservable | async">
      <td>{{ product.title }}</td>
      <td>{{ product.price }}</td>
      <td>
        <a [routerLink]="['/admin/products/', product.key]">Edit</a>
      </td>
    </tr>`

Reply:
Implemented the documentToDomainObject() helper function.